### DISCUSSION TITLE
Tycho Pomless and different surefire configurations?

### DISCUSSION THREAD

**Discussion by guw** [eclipse-tycho/tycho#2438] [Q&A] [answered]
Is it possible to have different configuration for the Tycho Surefire execution in a pomless setup? 

For example, tests in "...core.tests" are headless plug-in tests, others in "ui.tests" need the UI and yet others are pure Maven Surefire tests (no OSGi environment needed).

**Comment by laeubi** [comment, marked as answer]
Some different ways are documented here and you can of course combine them (please propose a PR if something is unclear/missing):

https://tycho.eclipseprojects.io/doc/master/TestingBundles.html

the important part is that you define some kind of filter so the different executions can distinguish what Tests are relevant, e.g. you can choose  the standard test pattern for "plain" tests and use another prefix for others, this is not yet in the documentation but if you can provide a Demo and description it would be great, there is an integration test showing this pattern:

https://github.com/eclipse-tycho/tycho/blob/master/tycho-its/projects/surefire.combinedtests/bundle.test/pom.xml

Beside that, in pomless mode it is usually good to define some properties in the root pom.xml (e.g. `<skipSureFire>true</skipSureFire>` or `<requireUI>false</requireUI>`) that is then used in the general execution configuration, and then you can se these flags in the build.properties as described here:

https://github.com/eclipse-tycho/tycho/wiki/Tycho-Pomless#define-properties

**Reply by guw** [reply]
Thanks @laeubi. That's the doc I was looking for but Google didn't rank high enough.
